IdentityCore/src/util/NSURL+MSIDExtensions.m

```diff
@@ -198,11 +198,10 @@ + (NSURL *)msidAddParameters:(NSDictionary<NSString *, NSString *> *)parameters
             MSID_LOG_WARN_PII(nil, @"Ignoring key: %@ value: %@", key, value);
             continue;
         }
-        __auto_type item = [[NSURLQueryItem alloc] initWithName:key value:value];
-        [queryItems addObject:item];
+        [queryDic setValue:value forKey:key];
     }
     
-    urlComponents.queryItems = queryItems;
+    urlComponents.queryItems = [(NSDictionary *)queryDic urlQueryItemsArray];
```

Comment:
nit: why do you need to cast to (NSDictionary *)?

Reply:
because we only have `NSDictionary+MSIDExtensions.h` but not `NSMutableDictionary+MSIDExtensions.h` :)